src/Stride.CommunityToolkit/Engine/GameExtensions.cs

```diff
@@ -64,6 +64,39 @@ async Task RootScript()
         Scene GetRootScene() => game.SceneSystem.SceneInstance.RootScene;
     }
 
+    /// <summary>
+    /// Initializes the game, starts the game loop, and handles game events.
+    /// </summary>
+    /// <remarks>
+    /// This method performs the following actions:
+    /// 1. Schedules the root script for execution.
+    /// 2. Initiates the game loop by calling <see cref="GameBase.Run(GameContext)"/>.
+    /// 3. Invokes the provided <paramref name="start"/> and <paramref name="update"/> delegates.
+    /// </remarks>
+    /// <param name="game">The Game instance to initialize and run.</param>
+    /// <param name="context">Optional GameContext to be used. Defaults to null.</param>
+    /// <param name="start">Optional action to execute at the start of the game. Takes the game as a parameter.</param>
+    /// <param name="update">Optional action to execute during each game loop iteration. Takes the game as a parameter.</param>
+    public static void Run(this Game game, GameContext? context = null, Action<Game>? start = null, Action<Game>? update = null)
```

Comment:
Any reason for this new extension? I checked your Example06 and you should be able to do the same with the original `Run()`?

```csharp
using var game = new Game();
game.Run(start: Start);

void Start(Scene rootScene)
{
    game.SetupBase3DScene();
    var entity = game.CreatePrimitive(PrimitiveModelType.Cube);
    entity.Transform.Position = new Vector3(1f, 0.5f, 3f);
    entity.Scene = rootScene;
}
```

- `game` is already available inside the `Start` method
- `rootScene` is also available

Reply:
Ah sorry I guess I can revert that.
I mostly try to avoid closures where possible because of unnecessary display classes created by the compiler.
Mostly personal preference I guess.
The methods I allow static delegates and I much prefer this as this allows you to move the methods to a static class instead of a local function.

Reply:
I am for any improvements. Not sure if I understand what you are saying but keep this extension there as well so I can see myself I guess 🙂